File: re-ssim-service/src/main/java/com/shinonometn/re/ssim/service/courses/repository/TermInfoRepository.java
package com.shinonometn.re.ssim.service.courses.repository;

import com.shinonometn.re.ssim.service.courses.entity.TermInfoEntity;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

public interface TermInfoRepository extends MongoRepository<TermInfoEntity, String> {
    Optional<TermInfoEntity> findByName(String name);

    Optional<TermInfoEntity> findByCode(String key);
}

File: re-ssim-components/re-ssim-data-kingo-application/src/main/java/com/shinonometn/re/ssim/data/kingo/application/repository/TermInfoRepository.kt
package com.shinonometn.re.ssim.data.kingo.application.repository

import com.shinonometn.re.ssim.data.kingo.application.entity.TermInfo
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface TermInfoRepository : JpaRepository<TermInfo, Int> {
    fun findByIdentity(get: String): Optional<TermInfo>

    @Query("select ti from TermInfo ti where ti.endDate is not null and ti.endDate >= ?1")
    fun findByDateAfter(current: Date): Collection<TermInfo>
}